### PR TITLE
Fixed deprecation warning when starting OSX console app

### PR DESCRIPTION
**Pull request by tbass**
This is the pull request for issue #749
Chnaged the deprecated way to the new TransformProcessType

### CODE REVIEW

**Comment by goobliata** [comment]
Could you fix the indentation? We indent with 3 spaces per stop. Thanks.

**Comment by goobliata** [comment]
Also does this actually fix any deprecation or just hide them? Do you get console output from the relevant examples?

**Comment by tbass** [comment]
Fixed the indention
It really fixes the deprecation warning when starting a console application and uses the proposed Apple way using ```TransformProcessType``` introduced in OSX 10.3. 
Relevant output is in issue #749

**Comment by goobliata** [comment]
And are you still getting console output?

**Comment by goobliata** [comment]
I tried it myself. Indeed it fixes the deprecations warnings, but for me windows now pop up behind everything else. Not for you?

**Comment by goobliata** [comment]
```
diff --git a/src/macosx/system.m b/src/macosx/system.m
index b441571..b8282f9 100644
--- a/src/macosx/system.m
+++ b/src/macosx/system.m
@@ -66,7 +66,7 @@ static void osx_tell_dock(void)
 {
    ProcessSerialNumber psn = { 0, kCurrentProcess };
    TransformProcessType(&psn, kProcessTransformToForegroundApplication);
-   [NSApplication sharedApplication];
+   [[NSApplication sharedApplication] activateIgnoringOtherApps:YES];
 }
```

With this patch on top of yours the windows come to the front. Can you confirm?

**Comment by tbass** [comment]
Nope, I can not confirm. The ex_* examples I use still come up in the same manner both ways (with or without your addition). Also my own little test game pops up in front.

I think it won't hurt to add the ```activateIgnoringOtherApps:YES``` as it explicitly activates the app, even though, Apple does not recommend to use it (ref: http://apple.co/2ocKTbB)

**Comment by fatcerberus** [comment]
For a game (the usual Allegro use case), I personally don't mind the potential focus-stealing.  I vote for keeping the flag.

**Comment by goobliata** [comment]
If you could add my patch to the PR I will merge this, thanks!

**Comment by tbass** [comment]
added your patch

File: src/macosx/system.m [hunk 1]
```diff
@@ -66,7 +66,7 @@ static void osx_tell_dock(void)
 {
    ProcessSerialNumber psn = { 0, kCurrentProcess };
    TransformProcessType(&psn, kProcessTransformToForegroundApplication);
-   [NSApplication sharedApplication];
+   [[NSApplication sharedApplication] activateIgnoringOtherApps: YES];
```
Comment (fatcerberus) [inline src/macosx/system.m]: This is neither here nor there, but I just have to say that Objective-C is a very strange language. :-p

**Pull request merged**